custom_cartridge_master/cartridges/custom_sanika1/cartridge/controllers/wishlist.js

```diff
@@ -0,0 +1,191 @@
+'use strict';
+
+var server = require('server');
+var ProductListMgr = require('dw/customer/ProductListMgr');
+var ProductList=require('dw/customer/ProductList');
+var ProductMgr = require('dw/catalog/ProductMgr');
+var Transaction = require('dw/system/Transaction');
+var URLUtils = require('dw/web/URLUtils');
+var collection = require('*/cartridge/scripts/util/collections');
+
+function getWishlistForAuthenticatedUser(req, res) {
+    var collections = require('*/cartridge/scripts/util/collections');
+    var ProductMgr = require('dw/catalog/ProductMgr');
+    var ProductListMgr = require('dw/customer/ProductListMgr');
+    var ProductList = require('dw/customer/ProductList');
+    var productHelper = require('*/cartridge/scripts/helpers/productHelpers');
+    var showProductPageHelperResult = productHelper.showProductPage(req.querystring, req.pageMetaData);
+
+    var currentCustomer = req.currentCustomer.raw;
+    var wishlists = ProductListMgr.getProductLists(currentCustomer, ProductList.TYPE_WISH_LIST);
+    var wishlist = wishlists.length > 0 ? wishlists[0] : null;
+    var wishlistItems = wishlist ? wishlist.getProductItems() : [];
+    var updatedWishlistItems = [];
+    var iterator = wishlistItems.iterator();
+
+    while (iterator.hasNext()) {
+        var item = iterator.next();
+        var product = ProductMgr.getProduct(item.productID);
+        var selectedVariantAttributes = {};
+
+       
+        if (product.isVariant()) {
+            var variationModel = product.getVariationModel();
+            var productVariationAttributes = variationModel.getProductVariationAttributes();
+
+           
+            collections.forEach(productVariationAttributes, function (attribute) {
+                var attributeValue = variationModel.getSelectedValue(attribute);
+                if (attributeValue) {
+                    selectedVariantAttributes[attribute.getID()] = {
+                        displayName: attribute.getDisplayName(),
+                        selectedValue: attributeValue.getDisplayValue()
+                    };
+                }
+            });
+        }
+
+        updatedWishlistItems.push({
+            productListItem: item,
+            selectedVariantAttributes: selectedVariantAttributes 
+        });
+    }
+    res.render('wishlist/show', {
+        wishlistItems: updatedWishlistItems,
+        addToCartUrl: showProductPageHelperResult.addToCartUrl
+    });
+}
+server.get('ShowProducts', function (req, res, next) {
+
+    var currentCustomer = req.currentCustomer;
+    
+    if (currentCustomer.raw.authenticated) {
```

Comment:
in this condition if and else doing the same thing.. then why condition needed here